src/fu-install-task.c

```diff
@@ -92,6 +92,59 @@ fu_install_task_get_is_downgrade (FuInstallTask *self)
 	return self->is_downgrade;
 }
 
+static gboolean
+fu_install_task_check_verfmt (FuInstallTask *self,
+			      GPtrArray *verfmts,
+			      FwupdInstallFlags flags,
+			      GError **error)
+{
+	FwupdVersionFormat fmt_dev = fu_device_get_version_format (self->device);
+	g_autoptr(GString) version_format = g_string_new (NULL);
+
+	/* for the error message */
+	for (guint i = 0; i < verfmts->len; i++) {
+		XbNode *verfmt = g_ptr_array_index (verfmts, i);
+		const gchar *tmp = xb_node_get_text (verfmt);
+		g_string_append_printf (version_format, "%s;", tmp);
+	}
```

Comment:
If this is only used for the error and warning messages I'd think it makes sense to split this off to it's own static function to build and allocate this string in those scenarios.

Reply:
Agreed. Done.